src/go/rpk/pkg/cli/topic/consume.go

```diff
@@ -183,19 +202,44 @@ func (c *consumer) consume() {
 			if done {
 				return
 			}
-
 			pend, hasEnd := c.getPartitionEnd(p.Topic, p.Partition)
 			if hasEnd && pend < 0 {
 				return // reached end, still draining client
 			}
 
+			serdeCache := make(map[int]serde.Serde)
 			for _, r := range p.Records {
+				var toStdErr bool
+				if c.decodeKey || c.decodeVal {
```

Comment:
I think this conditional can be eliminated because both the key and val have dedicated conditionals just below -- this doesn't do much but adds a tab indent level

Reply:
Good point, removed :+1: